src/messagingService/index.js

```diff
@@ -9,39 +11,45 @@ export const checkForPermissions = async () => {
     try {
       await messaging().requestPermission()
     } catch (error) {
-      console.log(error.message)
+      throw new MessagingError(error.message)
     }
   }
 }
 
 export const saveFCMTokenForCurrentUser = async () => {
   if (messaging().hasPermission()) {
     const fcmToken = await messaging().getToken()
-
     if (fcmToken) {
       try {
+        await CacheService.setItem('fcmToken', fcmToken)
         const storeFCMTokenInUserGroup = firebase.functions().httpsCallable('storeFCMTokenInUserGroup')
 
         await storeFCMTokenInUserGroup({ fcmToken: fcmToken })
       } catch (error) {
-        console.log(error.message)
+        throw new MessagingError(error.message)
       }
     }
   }
 }
 
 export const deleteFCMTokenForCurrentUser = async () => {
-  if (await messaging().hasPermission()) {
-    const fcmToken = await messaging().getToken()
+  let fcmToken
 
-    if (fcmToken) {
-      try {
-        const removeFCMTokenInUserGroup = firebase.functions().httpsCallable('removeFCMTokenInUserGroup')
+  try {
+    fcmToken = await CacheService.getItem('fcmToken')
 
-        await removeFCMTokenInUserGroup({ fcmToken: fcmToken })
-      } catch (error) {
-        console.log(error.message)
-      }
+    await CacheService.removeItem('fcmToken')
+  } catch (error) {
+    fcmToken = null
+  }
+
+  if (fcmToken) {
+    try {
+      const removeFCMTokenInUserGroup = firebase.functions().httpsCallable('removeFCMTokenInUserGroup')
+      
+      await removeFCMTokenInUserGroup({ fcmToken: fcmToken })
```

Comment:
When you're creating an object that has a key with the same name of the variable that has the value being assigned, you can skip the value assignment: `await removeFCMTokenInUserGroup({ fcmToken })`